crates/ruff_server/src/lint.rs

```diff
@@ -69,37 +69,29 @@ pub(crate) fn check(
 ) -> DiagnosticsMap {
     let source_kind = query.make_source_kind();
     let settings = query.settings();
-    let document_path = query.file_path();
+    let document_path = query.virtual_file_path();
 
     // If the document is excluded, return an empty list of diagnostics.
-    let package = if let Some(document_path) = document_path.as_ref() {
-        if is_document_excluded_for_linting(
-            document_path,
-            &settings.file_resolver,
-            &settings.linter,
-            query.text_document_language_id(),
-        ) {
-            return DiagnosticsMap::default();
-        }
+    if is_document_excluded_for_linting(
+        &document_path,
+        &settings.file_resolver,
+        &settings.linter,
+        query.text_document_language_id(),
+    ) {
+        return DiagnosticsMap::default();
+    }
 
-        detect_package_root(
-            document_path
-                .parent()
-                .expect("a path to a document should have a parent path"),
-            &settings.linter.namespace_packages,
-        )
-        .map(PackageRoot::root)
-    } else {
-        None
-    };
+    let package = detect_package_root(
```

Comment:
Same here, we can use `file_path` here

Reply:
Sorry! I missed this, created https://github.com/astral-sh/ruff/pull/18914